azurelinuxagent/ga/env.py

```diff
@@ -80,6 +82,13 @@ def run(self):
         self.dhcp_handler.conf_routes()
         self.hostname = self.osutil.get_hostname_record()
         self.dhcp_id_list = self.get_dhcp_client_pid()
+        # Cleanup MDS firewall rule and ensure WireServer firewall rule is set
+        # before we query goal state (for agents migrating from MDS protocol WS rule is not set)
+        # We are setting firewall rules before thread is spun off to avoid race condition
+        # where we query goal state in update before setting firewall rules.
+        if conf.enable_firewall() and remove_metadata_server_firewall_rule:
+            self.osutil.remove_firewall(METADATA_SERVER_ENDPOINT, uid=os.getuid())
```

Comment:
always remove MDS firewall rule or is this check ok?

Reply:
Discussed with Norberto, IMDS has same IP but is not protected in public so it is ok to remove 169.254.169.254 from firewall rules (Azure does not have 169.254.169.254 in firewall rules).